core/src/save/player_data.rs

```diff
@@ -1 +1,33 @@
+use std::fs::File;
+use std::io::Read;
 
+use uuid::Uuid;
+
+pub const UNSET_GAMEMODE: i32 = -1;
+
+/// Represents the contents of a player data file.
+#[derive(Default, Debug, Clone, Serialize, Deserialize)]
+pub struct PlayerData {
+    #[serde(default = "default_gamemode")]
+    #[serde(rename = "playerGameType")]
+    pub gamemode: i32,
+}
+
+/// Set to a magic value when player data doesn't exist (first-time join)
+fn default_gamemode() -> i32 {
+    // FIXME: Doesn't seem to work for default (non-existing NBT files)
```

Comment:
I'm not sure if this is a limitation of the NBT deserialization library, but this seems like the way to do it according to the serde [docs](https://serde.rs/attr-default.html). Seems to default to `0` no matter what.

Reply:
Switched to using `None`, which makes it cleaner imo.